object/src/mem.rs

```diff
@@ -889,7 +889,7 @@ where
                 AttributeSelectorStep::Nested { tag, item } => {
                     let e = obj
                         .entries
-                        .get(tag)
+                        .get(&tag)
```

Comment:
This change doesn't seem related, and looks undesirable according to clippy.

Reply:
I looked closer into this and it seems that the latest version of rust-analyzer is showing a false positive E0308 in this position. `tag` in this context is already a `&Tag`, so it would not need a reference. This can be verified by running `cargo check` after this change.

```suggestion
                        .get(tag)
```

I filed an issue at rust-analyzer in the meantime. https://github.com/rust-lang/rust-analyzer/issues/16657